on_node_configured.sh

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env bash
+
+# https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#file-bash_strict_mode-md
+set -euo pipefail
+
+#####
+# Script arguments
+#####
+SLURM_JWT_KEY="$1"
+# GitHub release tag for UQLE CLI tool
+# CLI_TAG="$2"
+# GitHub OAuth token - should have read access to UQLE CLI releases, and the UQLE stack repository
+# MACHINE_USER_TOKEN="$3"
+# UQLE_API_HOST="$4"
+
+# global variables
+JWT_KEY_DIR=/var/spool/slurm.state
+JWT_KEY_FILE="$JWT_KEY_DIR"/jwt_hs256.key
+SLURMDBD_USER="slurm"
+SLURM_PASSWORD_FILE=/root/slurmdb.password
+
+. /etc/parallelcluster/cfnconfig
+
+echo "Node type: $cfn_node_type"
+
+function modify_slurm_conf() {
+    # add JWT auth and accounting config to slurm.conf
+    # /opt/slurm is shared via nfs, so this only needs to be configured on head node
+
+    cat >>/opt/slurm/etc/slurm.conf <<EOF
+# Enable jwt auth for Slurmrestd
+AuthAltTypes=auth/jwt
+# ACCOUNTING
+JobAcctGatherType=jobacct_gather/linux
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=localhost
+AccountingStoragePort=6819
+EOF
+}
+
+function configure_users_common() {
+    sysctl user.max_user_namespaces=15000
+    usermod --add-subuids 165536-231071 --add-subgids 165536-231071 slurm
+}
+
+function configure_users_head_node() {
+    configure_users_common
+
+    cat <<'EOF' | tee -a /home/centos/.bashrc /home/slurm/.bashrc
+# Set variables to avoid podman conflicts between nodes due to nfs-sharing of /home
+# See basedir-spec at https://specifications.freedesktop.org/
+
+base_xdg_dir=$(mktemp -qd /tmp/"$(id -u)"-xdg-XXXXXXXXXX)
+
+export XDG_RUNTIME_DIR="$base_xdg_dir"/.runtime
+export XDG_DATA_HOME="$base_xdg_dir"/.data
+export XDG_STATE_HOME="$base_xdg_dir"/.state
+export XDG_CACHE_HOME="$base_xdg_dir"/.cache
+export XDG_CONFIG_HOME="$base_xdg_dir"/.config
+
+unset base_xdg_dir
+
+for directory in {"$XDG_RUNTIME_DIR","$XDG_DATA_HOME","$XDG_STATE_HOME","$XDG_CACHE_HOME","$XDG_CONFIG_HOME"}; do
+    mkdir -p "$directory"
+done
+
+alias podman="podman --runroot=\$XDG_RUNTIME_DIR --root=\$XDG_DATA_HOME"
+EOF
+}
+
+function write_jwt_key_file() {
+    # set the jwt key
+    if [ "$SLURM_JWT_KEY" ]; then
```

Comment:
*non-blocking, query:* Is there a reason to do `if [ "$MYVARIABLE" ]` instead of `if [[ -n "$MYVARIABLE" ]]` or vice-versa? I usually do the latter just because that's what's listed on https://devhints.io/bash#conditionals but I don't know whether there are any functional differences

Reply:
I usually use `-z/-n`, because I use the same cheat sheet. Not sure if this was changed intentionally. Since I'm using strict mode I doubt it matters - but non-empty string tests are better for this scenario. I think I'll defer changing it until the next MR because these inputs are changed around a bit in the next branch

Reply:
Yeah that makes sense